addons/hr_holidays/models/hr_leave_stress_day.py

```diff
@@ -17,26 +17,45 @@ class StressDay(models.Model):
     color = fields.Integer(default=lambda dummy: randint(1, 11))
     resource_calendar_id = fields.Many2one('resource.calendar', 'Working Hours',
         default=lambda self: self.env.company.resource_calendar_id.id, domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    department_ids = fields.Many2many('hr.department', string="Departments")
 
     _sql_constraints = [
         ('date_from_after_day_to', 'CHECK(start_date <= end_date)', 'The start date must be anterior than the end date.')
     ]
 
     @api.model
-    def get_stress_days(self, start_date, end_date, resource_calendar_id=None):
-        resource_calendar_id = resource_calendar_id or self.env.user.employee_id.resource_calendar_id or self.env.company.resource_calendar_id
+    def get_stress_days(self, start_date, end_date, resource_calendar_ids=None):
```

Comment:
The method should become api.multi (aka classic odoo method) and use self to retrieve the employees, instead of an api.model using a mystic context (which is usually a bad practice)